src/main/java/com/splunk/hecclient/LoadBalancer.java

```diff
@@ -66,4 +116,80 @@ public void send(final EventBatch batch) {
     public int size() {
         return channels.size();
     }
+
+    private void run() {
+        log.debug("Running healthcheck at an interval of {} seconds", this.hecConfig.getlbPollInterval()/1000.0);
+            Enumeration<String> iter = indexerInfo.keys();
+            for (; iter.hasMoreElements();) {
+                String indexerToCheck = iter.nextElement();
+                try {
+                    indexerCheckResult(sendHealthCheckRequest(indexerToCheck), indexerToCheck);
+                } catch (Exception e) {
+                    indexerCheckResult(false, indexerToCheck);
+                    log.error("encountered exception when checking health of indexer {}, " +
+                            "this means indexer and its channels are removed from the loadbalancer", e.getMessage());
+                }
+            }
+
+    }
+
+    private synchronized void indexerCheckResult(boolean success, String indexer) {
+        log.debug("indexerInfo map has {} ", indexerInfo);
+        log.debug("discardedIndexers map has {} ", discardedIndexers);
+        log.debug("Channels List is : {}", channels);
+
+        if(success) {
+            if(discardedIndexers.contains(indexer)) {
+                log.info("healthcheck passed for {} indexer, adding this indexer and its channels to the loadbalancer, " +
+                        "this indexer was previously removed from loadbalancer", indexer);
+                // Add channels for this indexer as healthcheck passed
+                List<HecChannel> channelsToAdd = indexerInfo.get(indexer);
+                channelsToAdd.forEach((channel) -> channels.add(channel));
+                discardedIndexers.remove(indexer);
+                index = 0;
+            }
+        } else {
+            if(!discardedIndexers.contains(indexer)) {
+                log.info("healthcheck failed for {} indexer, removing this indexer and its channels from the loadbalancer", indexer);
+                // Remove channels for this indexer as healthcheck failed
+                List<HecChannel> channelsToRemove = indexerInfo.get(indexer);
```

Comment:
Rather than removing it, could we ban it for a while, for example, 1 minute?

Reply:
we have that kind of feature already, its called backpressure, this is specifically for dropping indxers